var documenterSearchIndex = {"docs":
[{"location":"","page":"Home","title":"Home","text":"CurrentModule = FastaProcessing","category":"page"},{"location":"#FastaProcessing","page":"Home","title":"FastaProcessing","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Documentation for FastaProcessing.","category":"page"},{"location":"","page":"Home","title":"Home","text":"","category":"page"},{"location":"","page":"Home","title":"Home","text":"Modules = [FastaProcessing]","category":"page"},{"location":"#FastaProcessing.combined_dna_dataset","page":"Home","title":"FastaProcessing.combined_dna_dataset","text":"dnadatasets: Vector of dnadata read_range: e.g. dataset A has 500 reads, dataset B has 1000 reads, get Dict(1=>1:500, 2=>501:1500) L: length of the sequence for all the datasets (yes, they must have the same length)\n\ncombined the dna reads from all datasets following the order of the input dnadatasets useful for      1. referencing the different crosslinking signitures (readrange)     2. generate a datamatrix that includes all of the sequences\n\n\n\n\n\n","category":"type"},{"location":"#FastaProcessing.multi_2_multi_dna_dataset","page":"Home","title":"FastaProcessing.multi_2_multi_dna_dataset","text":"multi2multidnadataset     multiple datasets of dna seuqneces, e.g.          dataset1: [s₁, s₂,...,sₙ]         dataset2: [t₁, t₂,...,tₘ]     and multiple label sets         labels1: [l₁, l₂,...,lₙ]         labels2: [l₁, l₂,...,lₘ]\n\nnote that each dataset and its corresponding label set \nmust have the same cardinality\n\n\n\n\n\n","category":"type"},{"location":"#FastaProcessing.multi_2_one_dna_dataset","page":"Home","title":"FastaProcessing.multi_2_one_dna_dataset","text":"multi2onednadataset     multiple datasets of dna seuqneces, e.g.          dataset1: [s₁, s₂,...,sₙ]         dataset2: [t₁, t₂,...,tₙ]     and one label set         labels: [l₁, l₂,...,lₙ]\n\nnote that all datasets and the label set must have the same cardinality\n\nThis can happen in splicing datasets, where we only want to \nmodel multiple (non-endogeneous) intronic regions and how \nthey affect gene-expression\n\n\n\n\n\n","category":"type"},{"location":"#FastaProcessing.get_all_reads-Tuple{combined_dna_dataset}","page":"Home","title":"FastaProcessing.get_all_reads","text":"get_all_reads\n\n\n\n\n\n","category":"method"},{"location":"#FastaProcessing.get_range_dataset-Tuple{Any}","page":"Home","title":"FastaProcessing.get_range_dataset","text":"example: dataset A has 500 reads dataset B has 1000 reads return Dict(1=>1:500, 2=>501:1500)\n\n\n\n\n\n","category":"method"},{"location":"#FastaProcessing.obtain_dna_heads_and_dna_reads-Tuple{String}","page":"Home","title":"FastaProcessing.obtain_dna_heads_and_dna_reads","text":"obtaindnaheadsanddna_reads\n\nreturn:\n    dna_heads: header of all the fasta sequences (Vector)\n    dna_reads: dna strings\n\n\n\n\n\n","category":"method"},{"location":"#FastaProcessing.trim_dna_reads!-Tuple{Vector{String}, Vector{String}, Int64}","page":"Home","title":"FastaProcessing.trim_dna_reads!","text":"trimdnareads! trim the dna reads on both sides by n bases;  get rid of reads that contain n or N in the shortened read\n\nreturn:\n    dna_heads: header of all the fasta sequences (Vector) that are kept\n    dna_reads: trimmed dna sequences\n    len: len of reads after trimming\n\n\n\n\n\n","category":"method"}]
}
